dartsim/src/SDFFeatures.cc

```diff
@@ -420,9 +421,20 @@ Identity SDFFeatures::ConstructSdfModelImpl(
   // TODO(anyone) This may not work correctly with nested models and will need
   // to be updated once multiple canonical links can exist in a nested model
   // https://github.com/ignitionrobotics/ign-physics/issues/209
-  auto [modelID, modelInfo] = this->AddModel( // NOLINT
-      {model, modelFrame, _sdfModel.CanonicalLinkName()}, worldID);
-
+  auto [modelID, modelInfo] = [&] {
+    if (isNested)
+    {
+      return this->AddNestedModel(  // NOLINT
+          {model, _sdfModel.Name(), modelFrame, _sdfModel.CanonicalLinkName()},
+          _parentID, worldID);
+    }
+    else
+    {
+      return this->AddModel(  // NOLINT
+          {model, _sdfModel.Name(), modelFrame, _sdfModel.CanonicalLinkName()},
+          _parentID);
```

Comment:
```suggestion
      return this->AddModel(  // NOLINT
          {model, _sdfModel.Name(), modelFrame, _sdfModel.CanonicalLinkName()},
          _parentID, worldID);
```
I believe the world ID parameter is missing here.

Reply:
It's not that `worldID` is missing, but rather `_parentID` should be replaced with `worldID`.

Reply:
After looking at the code further, it does appear to be correct, although I would urge some caution because the type of entity that `_parentID` refers to is conditional. It may refer to a world or a model depending on whether the model that's being constructed is nested. This may be a strong recipe for some maintenance confusion and a future bug creeping in.

Reply:
The logic is `_parentID` and `worldID` are the same if the model is **not** nested. But I agree it is confusing. I changed it to `worldID` in https://github.com/ignitionrobotics/ign-physics/pull/228/commits/212c941319d4a7227ef38ae6bdbf196f2cec576b.

> This may be a strong recipe for some maintenance confusion and a future bug creeping in.

I agree. FWIW, this was done to avoid code duplication between `ConstructSdfModel` and `ConstructSdfNestedModel` where unlike `ConstructEmptyNestedModel`, `ConstructSdfNestedModel` is a feature that can construct a model and it's nested models from `sdf::Model`. i.e, `ConstructSdfNestedModel` is the same as `ConstructSdfModel` but can also handle nested models if they exist in the input `sdf::Model`.